src/devices/video/v9938.cpp

```diff
@@ -1138,76 +1138,123 @@ void v99x8_device::mode_graphic1(uint32_t *ln, int line)
 
 void v99x8_device::mode_graphic23(uint32_t *ln, int line)
 {
-	pen_t fg, bg, pen;
-	int nametbl_addr, patterntbl_addr, colourtbl_addr;
-	int pattern, x, xx, line2, name, charcode,
-	colour, colourmask, patternmask, xxx;
+	const int colourmask = ((m_cont_reg[3] & 0x7f) * 8) | 7;
+	const int patternmask = ((m_cont_reg[4] & 0x03) * 256) | 0xff;
+	const int scrolled_y = (line + m_cont_reg[23]) & 255;
```

Comment:
Why not express the left shifts using the shift operator to make it clearer (`<< 3` and `<< 8`), and express the mask as hex (`& 0xff`)?